lib/addons/fluxcd/index.ts

```diff
@@ -9,6 +9,51 @@ import { FluxGitRepository } from "./gitrepository";
 import { FluxKustomization } from "./kustomization";
 import { KubernetesManifest } from 'aws-cdk-lib/aws-eks/lib/k8s-manifest';
 
+/**
+ * Options for a FluxCD repository
+ */
+export interface FluxCDGitRepo extends Omit<spi.GitOpsApplicationDeployment, "repository">{
+
+    /** 
+     * Repository to create source from
+     */
+    repository: Omit<spi.GitRepositoryReference, "path" | "name">
+  /** 
+  * Flux SecretRef */
+  fluxSecretRefName?: string;
+
+  /** 
+  * Internal for Flux sync.
+  * Default `5m0s` */
+  fluxSyncInterval?: string;
+
+    /**
+     * List of kustomizations to create */
+    kustomizations?: FluxKustomizationProps[];
+}
+
+export interface FluxKustomizationProps {

```

Comment:
would it make sense to move this interface and the repo interface to the related files?

Reply:
I clearly remember replying to this, but my comment does not show up. We keep contracts together as customers will need all of them exported. so rather than creating dependencies between files, i suggest that we keep all contracts related to the addon either in the index or in a separate file called contracts.ts, types.ts and such.